### ISSUE TITLE
Terragrunt project count mismatch causes fatal panic

### ISSUE THREAD

**Issue by RickyHolland** [infracost/infracost#1281]
I'm currently evaluating infracost as part of our stack and unfortunately, I'm running into the below issue. 

For context, the project/directory I'm running the breakdown command against contains multiple modules/subfolders that each contain their own terragrunt.hcl. Some of which I'm filtering out using `export INFRACOST_TERRAGRUNT_FLAGS="--terragrunt-exclude-dir helm-* --terragrunt-exclude-dir k8s-*".

If I run the same command against an individual module/subfolder everything is working as expected.
`
```
`$ infracost breakdown --path .`

Detected Terragrunt directory at .
  ✔ Running terragrunt run-all terragrunt-info
  ✔ Running terragrunt run-all plan
  ✔ Running terragrunt show for each project
  ✔ Calculating monthly cost estimate


Error: An unexpected error occurred

runtime error: index out of range [1] with length 1
goroutine 1 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.17.5/x64/src/runtime/debug/stack.go:24 +0x65
main.handleUnexpectedErr(0x0, {0x11ceea0, 0xc000040fc0})
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:170 +0x45
main.main.func1()
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:42 +0x69
panic({0x11ceea0, 0xc000040fc0})
        /opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1038 +0x215
github.com/infracost/infracost/internal/apiclient.(*DashboardAPIClient).AddRun(0xc0004a7a68, 0xc0002b4a80, {0xc00041c1b8, 0x1, 0xc8}, {{0x1423a0b, 0x3}, {0x0, 0x0}, {0x0, ...}, ...})
        /home/runner/work/infracost/infracost/internal/apiclient/dashboard.go:91 +0xb0e
main.runMain(0xc0002f6500, 0xc0002b4a80)
        /home/runner/work/infracost/infracost/cmd/infracost/run.go:166 +0xb97
main.breakdownCmd.func1(0xc0002f6500, {0x1424d17, 0x2, 0x2})
        /home/runner/work/infracost/infracost/cmd/infracost/breakdown.go:42 +0x15c
github.com/spf13/cobra.(*Command).execute(0xc0002f6500, {0xc00029f760, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002af400)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
        /home/runner/work/infracost/infracost/cmd/infracost/main.go:55 +0x116

Environment:
Infracost v0.9.16
```

**Comment by hugorut** [comment]
Hey @RickyHolland, thanks for reporting this 🙏 . I'm looking into this now and will get back to you ASAP once we've figured out what the issue is.

**Comment by RickyHolland** [comment]
Thanks Hugo. 

I've done some further troubleshooting, using an EKS module as an example, and the interesting thing is if that I run `infracost breakdown --path .` within that subfolder then everything is working as expected. 

I then moved back up a directory (into the project dir which contains multiple modules) and targeted the EKS folder directly again using `"export INFRACOST_TERRAGRUNT_FLAGS="--terragrunt-include-dir eks"`. So in theory only the EKS subfolder would be within scope. If I then run the `infracost breakdown --path .` command again then I receive the index out of range error. I've also verified that only the eks subfolder is being targeted using the debug log level.

**Comment by hugorut** [comment]
@RickyHolland thanks for the additional info. Is it possible to try running the problem `breakdown` with `INFRACOST_PARALLELISM=1` env var set and letting me know if this solves the runtime panic.

**Comment by RickyHolland** [comment]
@hugorut Just gave the parallelism env var a go and unfortunately it still has the same error.

**Comment by hugorut** [comment]
@RickyHolland thanks, no worries, the search continues!!

**Comment by hugorut** [comment]
@RickyHolland - just an update, I think I've found the root cause of this but need to do some validation with examples locally. However, I'm stepping into a meeting for a couple hours. I'll update with progress after that.

**Comment by hugorut** [comment]
Hey @RickyHolland so we think we've managed to reproduce this locally and have identified the issue. Can you confirm that running 

```
infracost configure set enable_dashboard false
```

or 

```
INFRACOST_ENABLE_DASHBOARD=false
```

And running the breakdown cmd fixes the panic?

**Comment by RickyHolland** [comment]
Hey @hugorut,

That's done the trick, we are now able to view cost for multiple modules :).

**Comment by hugorut** [comment]
![](https://media.giphy.com/media/l0MYt5jPR6QX5pnqM/giphy.gif)

@RickyHolland thanks for your patience. We'll have a permanent fix for this issue out in the next release. I'll post here when it's out.

**Comment by hugorut** [comment]
@RickyHolland we're merged a fix into master for ready for the v0.9.17 release. I'll drop you a message when that version goes live.

**Comment by hugorut** [comment]
hey @RickyHolland a fix for this is now live as part of the `v0.9.17` release
